node/network/statement-distribution/src/lib.rs

```diff
@@ -1341,8 +1346,26 @@ async fn handle_incoming_message_and_circulate<'a>(
 		// that require dependents. Thus, if this is a `Seconded` statement for a candidate we
 		// were not aware of before, we cannot have any dependent statements from the candidate.
 		let _ = metrics.time_network_bridge_update_v1("circulate_statement");
+
+		let session_index = runtime.get_session_index_for_child(ctx.sender(), relay_parent).await;
+		let topology = match session_index {
+			Ok(session_index) => topology_storage.get_topology_or_fallback(session_index),
+			Err(e) => {
+				gum::debug!(
+					target: LOG_TARGET,
+					%relay_parent,
+					"cannot get session index for the specific relay parent: {:?}",
+					e
+				);
+
+				topology_storage.get_current_topology()
+			},
+		};
+		let required_routing = topology.required_routing_by_peer_id(peer, false);
```

Comment:
This is getting the `RequiredRouting` based not on the _originator_ of the message (the validator who signed the message) but the peer which sent us the message.

Reply:
I think this could possibly lead to misbehaviors where statements are sent before seconded messages.

Reply:
This should be fixed now I think.